app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+def cache(func: callable) -> callable:
+    results = {}
+
+    def inner(*args, **kwargs) -> callable:
+        key = args + tuple(sorted(kwargs.items()))
+        if key in results:
+            print("Getting from cache")
+            return results[key]
+        else:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            results[key] = result
+            return result
```

Comment:
gj, but try to use only one return for ur solution, cause u are returning the same thing twice )

Reply:
i am closed this branch can you go to develop5  the main content now there